base/cvd/cuttlefish/host/commands/cvd/instances/instance_database_test.cpp

```diff
@@ -443,23 +497,28 @@ TEST_F(CvdInstanceDatabaseTest, UpdateInstances) {
   }
   auto& db = GetDb();
 
-  cvd::InstanceGroup grp;
-  grp.set_home_directory(Workspace() + "/grp1_home");
-  grp.set_name("grp1");
-  auto ins1 = grp.add_instances();
-  ins1->set_name("ins1");
-  ins1->set_state(cvd::INSTANCE_STATE_PREPARING);
-  auto ins2 = grp.add_instances();
-  ins2->set_name("ins2");
-  ins2->set_state(cvd::INSTANCE_STATE_PREPARING);
-
-  auto add_res = db.AddInstanceGroup(grp);
+  InstanceGroupParams grp{
+      .group_name = "grp1",
+      .instances =
+          {
+              {
+                  .per_instance_name = "ins1",
+                  .initial_state = cvd::INSTANCE_STATE_PREPARING,
+              },
+              {
+                  .per_instance_name = "ins2",
+                  .initial_state = cvd::INSTANCE_STATE_PREPARING,
+              },
+          },
+  };
+
+  Result<LocalInstanceGroup> group_res =
+      LocalInstanceGroup::Create(grp).value();
+  auto add_res = db.AddInstanceGroup(*group_res);
```

Comment:
Please validate that `group_res` is successful before accessing the success value. Ideally with
```c++
ASSERT_THAT(group_res, IsOk());
```